proxy/http/HttpTransactHeaders.cc

```diff
@@ -421,14 +421,14 @@ HttpTransactHeaders::calculate_document_age(ink_time_t request_time, ink_time_t
 }
 
 bool
-HttpTransactHeaders::does_server_allow_response_to_be_stored(HTTPHdr *resp)
+HttpTransactHeaders::does_server_allow_response_to_be_stored(HTTPHdr *resp, bool ignore_no_store_and_no_cache_directives)
 {
-  uint32_t cc_mask = (MIME_COOKED_MASK_CC_NO_CACHE | MIME_COOKED_MASK_CC_NO_STORE | MIME_COOKED_MASK_CC_PRIVATE);
+  uint32_t cc_mask = MIME_COOKED_MASK_CC_PRIVATE | (ignore_no_store_and_no_cache_directives ? 0 : MIME_COOKED_MASK_CC_NO_STORE);
```

Comment:
Note that removing `MIME_COOKED_MASK_CC_NO_CACHE` is intentional here because, per the RFC, `no-cache` doesn't mean that the proxy can't store the response (that's what `no-store` means), rather it means that future requests for the object has to revalidate it with the origin before sending the cached response to the client.